packages/gatsby-plugin-image/src/components/intersection-observer.ts

```diff
@@ -1,12 +1,27 @@
+/* eslint-disable no-unused-expressions */
 import { RefObject } from "react"
 
 let intersectionObserver: IntersectionObserver
 
 type Unobserver = () => void
 
+const ioEntryMap = new WeakMap<HTMLElement, () => void>()
+/* eslint-disable @typescript-eslint/no-explicit-any  */
+const connection =
+  (navigator as any).connection ||
+  (navigator as any).mozConnection ||
+  (navigator as any).webkitConnection
+/* eslint-enable @typescript-eslint/no-explicit-any */
+
+// These match the thresholds used in Chrome's native lazy loading
```

Comment:
Please add a link to the resource
```suggestion
// These match the thresholds used in Chrome's native lazy loading
// @see https://web.dev/browser-level-image-lazy-loading/#distance-from-viewport-thresholds
```